src/python/pants/backend/docker/registries.py

```diff
@@ -26,37 +26,52 @@ def __init__(self, message):
 
 @dataclass(frozen=True)
 class DockerRegistryOptions:
-    address: str
+    address: str | None
     alias: str = ""
     default: bool = False
 
     @classmethod
-    def from_dict(cls, alias: str, d: dict[str, Any]) -> DockerRegistryOptions:
+    def from_dict(cls, alias: str, d: Mapping[str, Any]) -> DockerRegistryOptions:
+        if "address" not in d:
+            raise ValueError(f"Missing address option in Docker registry configuration {d!r}.")
+        address = d["address"] or None
+        if not isinstance(address, str) and address is not None:
+            raise ValueError(
+                f"Invalid Docker registry address, expected string but got {address!r}."
```

Comment:
I opted for not mentioning that a falsey value is accepted as well here, to keep the sentence brief and comprehensible. 
Also as it feels a bit like a niche feature, that you may find by accident or by reading the docs (when updated with this).